### ISSUE TITLE
Bug Report: struct negative filter removes all rows with empty fields (in the filtered column) from the set of results

### ISSUE THREAD

**Issue by Chris75forumname** [cosmocode/dokuwiki-plugin-struct#656]
Negative filtering should be conservative. It shoudn't remove unasked for items from the set of results. 
Struct negative filter removes all rows with empty fields from the set of results. I consider this a bug.

See forum discussion: 
https://forum.dokuwiki.org/d/21420-struct-checkbox-field-becomes-nul-when-unchecked-filter-fails

Related bug reports:
https://github.com/cosmocode/dokuwiki-plugin-struct/issues/358
https://github.com/cosmocode/dokuwiki-plugin-struct/issues/380

**Comment by splitbrain** [comment]
Please add all necessary info to reproduce the bug to the bug report. Having to try extracting the actual info from a long winding forum thread is not helpful.

**Comment by Chris75forumname** [comment]
Okay, in other words: Generally, in case of negative filtering, like so:

```
---- struct serial ----
schema  :  test
cols        :  *
filter       :  Foo != bar
csv         :  0
----
```

... provided, this filter comparator is working properly, only rows with the term "bar" in column Foo should be removed from the set. Rows with empty cell in column Foo should NOT be removed, because this was NOT requested.

**Comment by annda** [comment]
Fixed in 521b204
